src/Definition/RuntimeClassDefinition.php

```diff
@@ -44,12 +44,23 @@ public function hasMethod($name, $inherit = false)
     }
 
     /**
-     * @param $name
+     * @param string $name
+     * @param bool $inherit
      * @return bool
      */
-    public function hasConst($name)
+    public function hasConst($name, $inherit = false)
     {
-        return $this->reflection->hasConstant($name);
+        if (!$this->reflection->hasConstant($name)) {
+            return false;
+        }
+
+        if ($inherit) {
```

Comment:
I think the current code is correct. I added a few comments to make it clearer :)